src/amp-story-player/amp-story-player-impl.js

```diff
@@ -789,6 +795,10 @@ export class AmpStoryPlayer {
 
     this.currentIdx_ = storyIdx;
     this.onNavigation_();
+
+    if (pageId != null) {
```

Comment:
Having two calls to `goToPageId_()` in two different spots will make this code hard to follow. Can we wrap the logic to switch to a different story to another method and put that inside a conditional instead?